### PR TITLE
[Doc] Add cooldown by partitions

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

File: docs/sql-reference/sql-statements/data-definition/ALTER TABLE.md [hunk 1]
```diff
@@ -112,22 +112,25 @@ DROP TEMPORARY PARTITION <partition_name>
...
 
-1. The following properties of a partition can be modified:
+  - storage_medium
+  - storage_cooldown_ttl or storage_cooldown_time
+  - replication_num
 
-   - storage_medium
-   - storage_cooldown_time
-   - replication_num
+- For the table has only one partition, partition name is the same as the table name. If the table is divided into multiple partitions, you can use `(*)`to modify the properties of all partitions. which is more convenient.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/ALTER TABLE.md:131]: image
-> that has

Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/ALTER TABLE.md:131]: image
-> the partition name

Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/ALTER TABLE.md:131]: image
should be a comma (,)

File: docs/sql-reference/sql-statements/data-definition/ALTER TABLE.md [hunk 2]
```diff
@@ -325,7 +328,7 @@ You can add a generated column and specify its expression. [The generated column
 
 #### Modify table properties
 
-Currently, StarRocks supports modifying bloomfilter columns, colocate_with property, dynamic_partition property, enable_persistent_index property, replication_num property and default.replication_num property.
+Currently, StarRocks supports modifying bloomfilter columns, colocate_with property, dynamic_partition property, enable_persistent_index property, replication_num property and default.replication_num property, storage_cooldown_ttl property and storage_cooldown_time property.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/ALTER TABLE.md:331]: image
Wrap all these properties in code format.

File: docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md [hunk 3]
```diff
@@ -376,34 +376,78 @@ Since version 3.0, the primary key and sort key are decoupled in the Primary Key
...
+    If the FE configuration item `enable_strict_storage_medium_check` is set to `true`, the system strictly checks BE disk type when you create a table. If the storage medium you specified in CREATE TABLE is inconsistent with BE disk type, an error "Failed to find enough host in all backends with storage medium is SSD|HDD." is returned and table creation fails. If `enable_strict_storage_medium_check` is set to `false`, the system ignores this error and forcibly creates the table. However, cluster disk space may be unevenly distributed after data is loaded.<br>
+
+    From 2.3.6, 2.4.2, 2.5.1, and 3.0 onwards, the system automatically infers storage medium based on BE disk type if `storage_medium` is not explicitly specified.<br>
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md:399]: image
2.3.6, 2.4.2, 2.5.1, and 3.0 -> v2.3.6, v2.4.2, v2.5.1, and v3.0

File: docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md [hunk 4]
```diff
@@ -376,34 +376,78 @@ Since version 3.0, the primary key and sort key are decoupled in the Primary Key
...
+
+  - The system automatically sets this parameter to SSD in the following scenarios:
+
+    - The disk types reported by BEs (`storage_root_path`) contain only SSD.
+    - The disk types reported by BEs (`storage_root_path`) contain both SSD and HDD. Note that from 2.3.10, 2.4.5, 2.5.4, and 3.0 onwards, the system sets `storage_medium` to SSD when `storage_root_path` reported by BEs contain both SSD and HDD and the property `storage_cooldown_time` is specified.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md:404]: image
See the comment on Line 399.

File: docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md [hunk 5]
```diff
@@ -376,34 +376,78 @@ Since version 3.0, the primary key and sort key are decoupled in the Primary Key
...
+- When configuring these parameters, you need to specify `"storage_medium = "SSD"`.
+- If you do not configure these parameters, automatic storage cooldown is not be automatically performed.
+- Execute `SHOW PARTITIONS FROM <table_name>` to view the automatic storage cooldown time for each partition.
+
+**Limit**
+
+- Expression and List partitioning are not supported.
+- The partition column need to be of date type.
+- Multiple partition columns are not supported.
+- Primary key tables are not supported.
```
Comment (amber-create) [inline docs/sql-reference/sql-statements/data-definition/CREATE TABLE.md:440]: image
-> Primary Key

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by amber-create — APPROVED** [review]

**Review by EsoragotoSpirit — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.2

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**